code/controllers/subsystem/movement/move_handler.dm

```diff
@@ -76,10 +76,13 @@ SUBSYSTEM_DEF(move_manager)
 ///Adds a loop to our parent. Returns the created loop if a success, null otherwise
 /datum/movement_packet/proc/add_loop(datum/controller/subsystem/movement/subsystem, datum/move_loop/loop_type, priority, flags, datum/extra_info)
 	var/datum/move_loop/existing_loop = existing_loops[subsystem]
+
+	if(existing_loop?.compare_loops(arglist(args.Copy(2))))
+		return //it already exists stop trying to make the same moveloop
+
 	if(existing_loop && existing_loop.priority > priority)
 		if(!(existing_loop.flags & MOVEMENT_LOOP_IGNORE_PRIORITY) && !(flags & MOVEMENT_LOOP_IGNORE_PRIORITY))
```

Comment:
Should do the priority check first, since it checks close to the same thing and is a bit less wild

Reply:
While i would not say its the exact same but yeah  makes sense to do it first since the priority check doesn't need a whole new proc to work